src/main/java/org/wildfly/security/auth/util/GSSCredentialSecurityFactory.java

```diff
@@ -131,6 +131,8 @@ public static Builder builder() {
         private boolean debug;
         private boolean wrapGssCredential;
         private boolean checkKeyTab;
+        private long lastFailTime = 0;
```

Comment:
This probably should be volatile to behave as expected, because of concurrency. And not leave it handled by unpredictable behaviour of java memory model.